README.md

```diff
@@ -1,32 +1,504 @@
+```
+
+The remainder of the initialization process depends on how you want to request permission from the user to accept push notifications.
+
+### Timing the device token (push permission) request
+
+When a user first installs a push-enabled app, the latest device token for push notifications is requested, which triggers a permission request to the user to accept push notifications from the app. Swrve has built-in functionality to manage the timing of this request—you can time it to one of the following options:
+
+* Request permission at the start of the session. This is the default configuration.
+* Request permission when one of a list of events is triggered.
+
+For more information about uploading device tokens, see [How Do I Upload Push Device Tokens?](http://docs.swrve.com/faqs/push-notifications/upload-push-device-tokens/)
+
+Apple only asks for permission to receive push notifications once per app, under normal circumstances. As a result, when you want to test your push notification integration on a QA device, you must reset the iOS push permissions to test the different timing configurations for push permissions outlined above. For more information, see [How Do I Reset iOS Permissions?](http://docs.swrve.com/faqs/push-notifications/reset-ios-permissions-for-push-notifications/)
+
+1. Request Push Permission at Start of Session (Default)
+
+  By default, the Swrve SDK requests push permission from the user on their first session. This is the simplest option and requires no additional code changes.
+
+2. Request Push Permission when Certain Events are Triggered
+
+  If you want to request push permission upon the triggering of certain events, add the event names to SwrveConfig.pushNotificationEvents when you initialize the SDK.
+
+  ```
+  // Initialize Swrve with custom events that will trigger the push notification dialog
+  SwrveConfig* config = [[SwrveConfig alloc] init];
+  config.pushEnabled = YES;
+  config.pushNotificationEvents = [[NSSet alloc] initWithArray:@[@"level_up", @"battle_win"]];
+  [Swrve sharedInstanceWithAppID:your_app_id apiKey:@"your_api_key" config:config launchOptions:launchOptions];
+  // Initialize Swrve with custom events that will trigger the push notification dialog
+  ```
+
+  If a user never triggers these events, they will never be available for push notifications. So select events that are most likely to be triggered by all but the most transient of users.
+
+### Processing custom payload
+
+If you want to capture any custom key/value pairs passed with the push notification, capture the notification data in both of your application’s `didFinishLaunchingWithOptions` and `didReceiveRemoteNotification` methods this:
+
+```
+- (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+   // Capture the notification data that opened the app
+   NSDictionary * remoteNotification = [launchOptions objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey];
```

Comment:
probably worth mentioning that the SDK should be initialized before dealing with the notification